storage/postgreskv/client.go

```diff
@@ -4,90 +4,92 @@
 package postgreskv
 
 import (
+	"bytes"
 	"context"
 	"database/sql"
 
 	"github.com/lib/pq"
 	"github.com/zeebo/errs"
-	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+	"gopkg.in/spacemonkeygo/monkit.v2"
 
 	"storj.io/storj/private/dbutil"
+	"storj.io/storj/private/dbutil/pgutil"
+	"storj.io/storj/private/dbutil/txutil"
 	"storj.io/storj/storage"
 	"storj.io/storj/storage/postgreskv/schema"
 )
 
 const (
-	defaultBatchSize = 10000
-	defaultBucket    = ""
+	defaultBatchSize          = 128
+	defaultRecursiveBatchSize = 10000
 )
 
 var (
 	mon = monkit.Package()
 )
 
-// Client is the entrypoint into a postgreskv data store
+// Client is the entrypoint into a postgreskv2 data store
 type Client struct {
-	URL    string
-	pgConn *sql.DB
+	db *sql.DB
 }
 
-// New instantiates a new postgreskv client given db URL
+// New instantiates a new postgreskv2 client given db URL
 func New(dbURL string) (*Client, error) {
-	pgConn, err := sql.Open("postgres", dbURL)
+	dbURL = pgutil.CheckApplicationName(dbURL)
+
+	db, err := sql.Open("postgres", dbURL)
 	if err != nil {
 		return nil, err
 	}
 
-	dbutil.Configure(pgConn, mon)
-
-	// TODO: this probably should not happen in constructor
-	err = schema.PrepareDB(context.TODO(), pgConn, dbURL)
+	dbutil.Configure(db, mon)
+	//TODO: Fix the parameters!!
```

Comment:
This needs to be done still

Reply:
? what parameters need fixing?

Reply:
Oh, thats also done.

Reply:
I remove it a second PR, to get this merged.